components/ai_chat/core/browser/conversation_driver.cc

```diff
@@ -257,22 +257,32 @@ bool ConversationDriver::MaybePopPendingRequests() {
     return false;
   }
 
-  if (!pending_request_) {
+  if (!pending_conversation_entry_) {
     return false;
   }
 
-  mojom::ConversationTurn request = std::move(*pending_request_);
-  pending_request_.reset();
+  // We don't discard requests related to summarization until we have the
+  // article text.
+  if (article_text_.empty() && pending_message_needs_page_content_) {
+    return false;
+  }
+
+  mojom::ConversationTurn request = std::move(*pending_conversation_entry_);
+  pending_conversation_entry_.reset();
+  pending_message_needs_page_content_ = false;
   MakeAPIRequestWithConversationHistoryUpdate(std::move(request));
   return true;
 }
 
 void ConversationDriver::MaybeGeneratePageText() {
   const GURL url = GetPageURL();
 
+  // Call this observer in-case we early return
+  // ex. a listener in the UI might expect siteInfo or its related property to
+  // change
+  OnPageHasContentChanged(false);
```

Comment:
since the design and scope of the toggle UI have changed so much within this PR, there is no need for this observer call anymore. i have tested it without it, and i don't see any weird UI bugs.

removed.